### PR TITLE
fix: use GH token for podman packages download

### LINKED ISSUE #416: build is fetching github releases without having a way to provide a GITHUB_TOKEN
### Bug description

When building podman desktop, we search for latest binaries on containers/podman asking `api.github.com`

But the Octokit used in the js script is not using/providing GITHUB_TOKEN that we have when running in a github action, leading to rating limit being hit

### Operating system

macOS builds

### Version

0.0.5

### Steps to reproduce

https://github.com/containers/podman-desktop/actions/runs/2881574264

### Relevant log output

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by evidolob**
### What does this PR do?
Use GitHub token in octokit while downloading podman packages.

### Screenshot/screencast of this PR
n/a

### What issues does this PR fix or reference?
Fix https://github.com/containers/podman-desktop/issues/416

### CODE REVIEW

**Comment by benoitf** [comment]
Hi @evidolob is that GITHUB_TOKEN automatically exported to the scripts ? or we need to add it in the yaml ?
I thought we have access to secrets.GITHUB_TOKEN but for the env, it requires an additional export

**Review by benoitf — APPROVED** [review]
I think we'll need the token on release workflow as well

File: .github/workflows/pr-check.yaml [hunk 1]
```diff
@@ -24,6 +24,8 @@ jobs:
     name: Windows
     runs-on: windows-2022
     timeout-minutes: 60
+    env:
+      GITHUB_TOKEN: ${{ github.token }}
```
Comment (benoitf) [inline .github/workflows/pr-check.yaml]: it's not secrets.something ?

Comment (benoitf) [reply .github/workflows/pr-check.yaml]: https://docs.github.com/en/actions/security-guides/automatic-token-authentication#example-1-passing-the-github_token-as-an-input 

```
${{ secrets.GITHUB_TOKEN }}
```

Comment (evidolob) [reply .github/workflows/pr-check.yaml]: It is also in context https://docs.github.com/en/actions/learn-github-actions/contexts#github-context

Comment (evidolob) [reply .github/workflows/pr-check.yaml]: But I change to `GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}` to unify the usage

**Review by benoitf — APPROVED** [review]
let's merge 👍

**Pull request merged**